app/controllers/password_resets_controller.rb

```diff
@@ -32,28 +31,27 @@ def update
   end
 
   private
-    def user_params
-      params.require(:user).permit(:password, :password_confirmation)
-    end
 
-    def get_user
-      @user = User.find_by(email: params[:email])
-    end
+  def user_params
+    params.require(:user).permit(:password, :password_confirmation)
+  end
 
-    # Confirms a valid user.
-    def valid_user
-      unless (@user && @user.activated? &&
-              @user.authentic_token?(:reset, params[:id]))
-        flash[:alert] = t('user_mailer.password_reset.invalid')
-        redirect_to root_url
-      end
-    end
+  def get_user
+    @user = User.find_by(email: params[:email])
+  end
 
-    # Checks expiration of reset token.
-    def check_expiration
-      if @user.password_reset_expired?
-        flash[:alert] = t('user_mailer.password_reset.expired_token')
-        redirect_to new_password_reset_url
-      end
-    end
+  # Confirms a valid user.
+  def valid_user
+    return if !@user || !@user.activated? ||
```

Comment:
Rewording changed the logic here. Password reset flow will not work for valid users. Should it be `return unless ...` instead?

Reply:
Also, fixing this will brake some test cases for password reset controller. See comments below.